src/lib/services/export-import-service.ts

```diff
@@ -95,24 +93,70 @@ export default class ExportImportService {
         this.segmentStore = stores.segmentStore;
         this.flagResolver = flagResolver;
         this.featureToggleService = featureToggleService;
+        this.contextFieldStore = stores.contextFieldStore;
         this.logger = getLogger('services/state-service.js');
     }
 
-    async export(query: ExportQuerySchema): Promise<IExportData> {
-        const [features, featureEnvironments, featureStrategies] =
-            await Promise.all([
-                this.toggleStore.getAllByNames(query.features),
-                (
-                    await this.featureEnvironmentStore.getAll({
-                        environment: query.environment,
-                    })
-                ).filter((item) => query.features.includes(item.featureName)),
-                this.featureStrategiesStore.getAllByFeatures(
-                    query.features,
-                    query.environment,
+    async export(
+        query: ExportQuerySchema,
+        userName: string,
+    ): Promise<IExportData> {
+        const [
+            features,
+            featureEnvironments,
+            featureStrategies,
+            strategySegments,
+            contextFields,
+            featureTags,
+        ] = await Promise.all([
+            this.toggleStore.getAllByNames(query.features),
+            await this.featureEnvironmentStore.getAll((builder) => {
```

Comment:
I would love some feedback on this. Most of our `getAll` methods allows us build queries outside store like this.

**Pros:** Dont need to create a separate store function every time we want to run a different query on dataset.
**Cons:** We write Knex in service level.